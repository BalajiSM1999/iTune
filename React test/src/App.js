import MainPage from './components/main';
import './style/App.css';
import'./style/responsive.css'

function App() {
  return (
    <div className="App">
      <MainPage />
    </div>
  );
}

export default App;
