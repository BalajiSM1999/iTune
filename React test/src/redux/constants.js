export const PRODUCT_LIST_REQUEST = 'PRODUCT_LIST_REQUEST';
export const PRODUCT_LIST_SUCCESS = 'PRODUCT_LIST_SUCCESS';
export const PRODUCT_LIST_FAIL = 'PRODUCT_LIST_FAIL';

export const PRODUCT_SORT_REQUEST = 'PRODUCT_SORT_REQUEST';
export const PRODUCT_SORT_SUCCESS = 'PRODUCT_SORT_SUCCESS';
export const PRODUCT_SORT_FAIL = 'PRODUCT_SORT_FAIL';