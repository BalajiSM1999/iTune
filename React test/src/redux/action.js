import axios from "axios";
import { PRODUCT_LIST_FAIL, PRODUCT_LIST_REQUEST, PRODUCT_LIST_SUCCESS, PRODUCT_SORT_FAIL, PRODUCT_SORT_REQUEST, PRODUCT_SORT_SUCCESS } from "./constants";

export const listProducts = () => async (dispatch) =>{
    dispatch({
        type: PRODUCT_LIST_REQUEST
    });
    try{
        const {data} = await axios.get("https://itunes.apple.com/us/rss/toppaidapplications/limit=100/json")
        dispatch({type: PRODUCT_LIST_SUCCESS, payload: data.feed.entry})
    }catch(error){
        dispatch({type: PRODUCT_LIST_FAIL, payload: error.message})
    }
}
export const sortProducts = ({product}) =>  (dispatch) =>{
    dispatch({
        type: PRODUCT_SORT_REQUEST
    });
    try{
        console.log(product)
        const data = ''
        dispatch({type: PRODUCT_SORT_SUCCESS, payload: product})
    }catch(error){
        dispatch({type: PRODUCT_SORT_FAIL, payload: error.message})
    }
}