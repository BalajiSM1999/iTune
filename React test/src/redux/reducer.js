import { PRODUCT_LIST_FAIL, PRODUCT_LIST_REQUEST, PRODUCT_LIST_SUCCESS, PRODUCT_SORT_FAIL, PRODUCT_SORT_REQUEST, PRODUCT_SORT_SUCCESS } from "./constants";

export const productListReducer = (state={products: [], error:'', loading: false}, action) =>{
    switch(action.type){
        case PRODUCT_LIST_REQUEST:
            return {loading: true};
            case PRODUCT_LIST_SUCCESS:
                return{loading :false, products: action.payload}
                case PRODUCT_LIST_FAIL :
                    return{loading: false, error:action.payload}
            default:
                return state;
    }
}
export const productSortReducer = (state=[], action) =>{
    switch(action.type){
        case PRODUCT_SORT_REQUEST:
            return {loading: true};
            case PRODUCT_SORT_SUCCESS:
                return{loading :false, products: action.payload}
                case PRODUCT_SORT_FAIL :
                    return{loading: false, error:action.payload}
            default:
                return state;
    }
}