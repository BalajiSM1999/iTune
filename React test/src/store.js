

import { applyMiddleware, combineReducers, compose, createStore } from 'redux';
import thunk from 'redux-thunk';
import { productListReducer, productSortReducer } from './redux/reducer';


const reducer = combineReducers({
    productList : productListReducer,
    productSort : productSortReducer
})



const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const store = createStore(reducer, composeEnhancer(applyMiddleware(thunk)));
export default store;


/*

import { configureStore } from "@reduxjs/toolkit";
import { setupListeners } from "@reduxjs/toolkit/query";
import { usersApi } from "./redux/products";

 const store = configureStore({
  reducer: {
    [usersApi.reducerPath]: usersApi.reducer,
  },
  middleware: (getDefaultMiddiware) =>
    getDefaultMiddiware().concat(usersApi.middleware),
});

setupListeners(store.dispatch);
export default store

*/