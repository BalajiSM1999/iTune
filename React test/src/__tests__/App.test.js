import App from "../App"
import renderer from "react-test-renderer"
import MainPage from "../components/main"
import {fireEvent, render } from "@testing-library/react"
import { listProducts } from "../redux/action"
import { Provider } from "react-redux"
import store from "../store"
import mockAxios from "axios";
import { productListReducer } from "../redux/reducer"


/*
it ("matches snapshot", ()=>{
    const tree= renderer.create(<Provider store={store}><App /></Provider>);
    expect(tree).toMatchSnapshot();
    })

    it('should render drop button', () => {
        const {queryByTitle} =render(<Provider store={store}><MainPage/></Provider> );
        const btn = queryByTitle("drop-btn");
        expect(btn).toBeTruthy()
      });

      test("should be able to view charts", () =>{
        const mockFn = jest.fn();
        const {getByTestId} =render(<Provider store={store}><MainPage handleSubmit={mockFn} /></Provider>);
        const button = getByTestId("btn");
        fireEvent.submit(button);
    
        expect(mockFn).toHaveBeenCalledTimes(0);
    })

   */

    describe('users Reducer', () => {
      const initialState = {
        loading: false,
        error: '',
        products: [],

      };
    
      it('returns the initial state when an action type is not passed', () => {
        const reducer = productListReducer(undefined, {});
    
        expect(reducer).toEqual(initialState);
      });

    it('handles PRODUCT_LIST_REQUEST as expected', () => {
      const reducer = productListReducer(initialState, { type: "PRODUCT_LIST_REQUEST" });
    
      expect(reducer).toEqual({
        loading: true,
      });
    });

    it('handles PRODUCT_LIST_SUCCESS as expected', () => {
      const reducer = productListReducer(initialState, { type: "PRODUCT_LIST_SUCCESS" });
    
      expect(reducer).toEqual({
        loading: false,
      });
    });
    it('handles PRODUCT_LIST_FAIL as expected', () => {
      const reducer = productListReducer(initialState, { type: "PRODUCT_LIST_FAIL" });
    
      expect(reducer).toEqual({
        loading: false,
      });
    });
  });


    /*

    it("handles PRODUCT_LIST_SUCCESS as expected", () => {
      const reducer = productListReducer(initialState, {
        type: "PRODUCT_LIST_SUCCESS",
        payload: {
          data: [
            {
              products:{
                [0]:{
                  ["im:name"]:{
                    label: "Minecraft"
                  }
                }
              }
            }
          ]
        }
      });
  
      expect(reducer).toEqual({
        products:{
          [0]:{
            ["im:name"]:{
              label: "Minecraft"
            }
          }
        },
        loading: false,
      });
    });


  */
 // https://medium.com/asos-techblog/how-to-test-your-react-redux-application-48d90481a253