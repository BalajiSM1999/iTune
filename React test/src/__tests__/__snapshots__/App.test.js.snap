// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
>
  <div>
    <button
      onClick={[Function]}
    >
      get
    </button>
    <div>
      <h1
        className="heading"
      >
        iTune Top Paid Applications
      </h1>
      <div
        className="charts"
      >
        <canvas
          height={400}
          onClick={[Function]}
          role="img"
          width={600}
        />
      </div>
    </div>
    <div>
      <div
        className="funcs"
      >
        <div
          className="search"
        >
          <p>
            Search: 
            <input
              id="searchTerm"
              onChange={[Function]}
              onKeyPress={[Function]}
              placeholder="filter by app name"
              type="text"
            />
          </p>
        </div>
        <select
          className="select"
          onChange={[Function]}
          onClick={[Function]}
        >
          <option
            value="[im:releaseDate].label"
          >
            Date
          </option>
          <option
            value="title.label"
          >
            Name
          </option>
          <option
            value="['im:price'].attributes.amount"
          >
            Price
          </option>
          <option
            value="category.attributes.label"
          >
            Category
          </option>
        </select>
      </div>
      <div
        id="divided"
      />
    </div>
  </div>
</div>
`;
